src/Database/Model.php

```diff
@@ -226,11 +226,10 @@ protected function bootNicerEvents()
                 }
 
                 self::$eventMethod(function ($model) use ($method) {
-                    $model->fireEvent('model.' . $method);
-
                     if ($model->methodExists($method)) {
-                        return $model->$method();
+                        $model->bindEvent('model.' . $method, [$model, $method]);
                     }
+                    $model->fireEvent('model.' . $method);
```

Comment:
I think we might have to make this event halting and return the result in order to keep compatibility. Also I think we have, but just to make sure, did we ever fix the issue with caching models causing issues if they had closures bound to events locally? Because if we didn't fix that issue then making this change would render all models uncacheable everywhere.

Reply:
> I think we might have to make this event halting and return the result in order to keep compatibility. Also I think we have, but just to make sure, did we ever fix the issue with caching models causing issues if they had closures bound to events locally? Because if we didn't fix that issue then making this change would render all models uncacheable everywhere.

@LukeTowers Wasn't the issue fixed with Laravel 9 with the serialized closures? Or are you talking about something else?

Reply:
That's what I'm talking about, I just don't remember if we actually fixed it for models or not. Are you able to test it out locally with this PR?

Reply:
It's been added to the Event\Dispatcher class for all event listeners using the `Support\Serialization::wrapClosure()` method.

Reply:
One of the PR that completed the work:
https://github.com/wintercms/storm/pull/69/files